fix: Bucket dates should be set to the start of the month

clockworked247 opened the pull request:
Closes #636

clockworked247 committed:
fix: Bucket dates should be set to the start of the month

Closes #636

laurelnaiad commented:
Is it relvevant whether we do 

```
var minDate = moment('2009-06-01T00:00:00Z').startOf('M').tz(spec.search.timezone);
```

vs.

```
var minDate = moment('2009-06-01T00:00:00Z').tz(spec.search.timezone).startOf('M');
```

? Just wondering. I don't know exactly how the .tz function works, but that latter seems to me to be maybe better? Get the value into the right timezone **before** jumping to the beginning of the month so that we don't go to the start of the wrong month? Or maybe it doesn't matter. If you're confident that it works as you have it, that's fine, just want to make sure.

clockworked247 committed:
browser: date range selection at edges correction added

closes marklogic/samplestack-internal#185

other small fixes includes:
 * label issue on chart when only one bar showing
 * placeholder end date when nothing selected off by one month

clockworked247 committed:
fix: reordered startOf() use

following the docs ordering as seen at: http://momentjs.com/timezone/docs/#/using-timezones/converting-to-zone/
I moved the use of "startOf" to after the use of timezone.

clockworked247 commented:
Checked the docs: http://momentjs.com/timezone/docs/#/using-timezones/converting-to-zone/
It seems they prefer the shifting of the moment to occur after the use of timezone.  I've made a commit to match their examples.

FYI, moving the startOf() call before or after the tz() call had no effect on the output.

laurelnaiad commented:
If it works the way I was guessing it would, it would only have an impact on about half of the world (based on whether you're plus or minus from GMT or something like that...)... thanks for looking into it!

clockworked247 commented:
@laurelnaiad Yea, only half... haha.  Good catch.  :+1:

laurelnaiad commented:
It may very well have been a non-issue. Timezones make my head hurt. If we're doing it like they do in the examples and it looks right, that's probably good enough. :)

laurelnaiad commented:
Just notice that the last commt never got merged. It is now a4c1153db82b39e999cc5399303ed208e905486a
